### PR TITLE
[SPARK-28920][INFRA] Set up java version for github workflow

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch adds java version parameter to GitHub workflow conf for JDK8/11.

### Why are the changes needed?

As we want to build JDK8/11 on GitHub workflow, we might need to add java version according current matrix.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

See the GitHub workflow run result.

### CODE REVIEW

**Comment by viirya** [comment]
cc @dongjoon-hyun @dbtsai

**Comment by viirya** [comment]
IIUC, we may need this parameter. If I understand it wrongly, please let me know, I will close this. Thanks.

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -24,4 +24,4 @@ jobs:
       run: |
         export MAVEN_OPTS="-Xmx2g -XX:ReservedCodeCacheSize=512m -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN"
         export MAVEN_CLI_OPTS="--no-transfer-progress"
-        ./build/mvn $MAVEN_CLI_OPTS -DskipTests package
+        ./build/mvn $MAVEN_CLI_OPTS -DskipTests -Djava.version=${{ matrix.java }} package
```
Comment (dongjoon-hyun) [inline .github/workflows/master.yml]: Ur, for GitHub workflow, we aims to do build-test with a single JDK. So, this is not required, isn't it?
Technically, when we build with JDK8 and test them on JDK11, we use `-Djava.version=11` during testing.

Comment (viirya) [reply .github/workflows/master.yml]: When we run mvn, its java.version property is 1.8, if not specified. Even we use matrix, I think mvn should still build with 1.8?

Comment (dongjoon-hyun) [reply .github/workflows/master.yml]: Oh, got it. You want to use `11` for `source` and `target` completely, right?

Comment (dongjoon-hyun) [reply .github/workflows/master.yml]: Actually, that one is my original intention. But this PR is better because it aims to move forward from it.

Comment (dongjoon-hyun) [reply .github/workflows/master.yml]: Previously, I wanted to use JDK11 for building and testing with `-release 8` option which means `source=8`, `target=8`, `bootclass=8`. It's an intermediate status.

Comment (dongjoon-hyun) [reply .github/workflows/master.yml]: It's interesting. Is there a way to have all of those combinations?

Comment (viirya) [reply .github/workflows/master.yml]: yeah, I think java.version controls source and target.

**Comment by dongjoon-hyun** [comment]
BTW, I love this PR, @viirya . Thanks. :)

**Comment by dongjoon-hyun** [comment]
I'll rebase and merge this after mine, @viirya .

**Comment by viirya** [comment]
@dongjoon-hyun thanks!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Oh.. Resovling conflicts seem not good. It removes the PR template and shows me as a co-author. Sorry about that, @viirya .

**Comment by viirya** [comment]
No problem at all! thanks @dongjoon-hyun

**Pull request closed without merging**